### PR TITLE
fix(security) sanitize contact groups

### PR DESCRIPTION
**Pull request by loiclau**
## Description

Available Cross-site Scripting (XSS) Stored/Persistent in "Configuration > Users > ContactGroups" form

Sanitize ContactGroups form

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 19.10.x
- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

see MON-5916

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by kduret — COMMENTED** [review]
Please update your PR description
If you don't fill some parts of the PR template, remove those

File: www/include/configuration/configObject/contactgroup/DB-Func.php [hunk 1]
```diff
@@ -190,26 +194,35 @@ function insertContactGroup($ret)
...
-        "', '" . CentreonDB::escape($ret["cg_comment"]) . "', '" . $ret["cg_activate"]["cg_activate"] . "')";
-    $pearDB->query($rq);
+    $stmt = $pearDB->prepare(
+        "INSERT INTO `contactgroup` (`cg_name`, `cg_alias`, `cg_comment`, `cg_activate`)
+        VALUES (:cgName, :cgAlias, :cgComment, :cgActivate)"
+    );
+
+    $stmt->bindValue(':cgName', $cgName, \PDO::PARAM_STR);
+    $stmt->bindValue(':cgAlias', $cgAlias, \PDO::PARAM_STR);
+    $stmt->bindValue(':cgComment', $cgComment, \PDO::PARAM_STR);
```
Comment (sc979) [inline www/include/configuration/configObject/contactgroup/DB-Func.php:209]: what will happen if comment contains only illegal chars ?
filter_var will return false and bindValue will try to bind false
same for other fields

File: www/include/configuration/configObject/contactgroup/DB-Func.php [hunk 2]
```diff
@@ -236,19 +253,25 @@ function updateContactGroup($cg_id = null, $params = array())
         $ret = $form->getSubmitValues();
     }
 
-    $ret["cg_name"] = $centreon->checkIllegalChar($ret["cg_name"]);
+    $cgName = $centreon->checkIllegalChar(filter_var($ret["cg_name"], FILTER_SANITIZE_STRING));
+    $cgAlias = filter_var($ret["cg_alias"], FILTER_SANITIZE_STRING);
+    $cgComment = filter_var($ret["cg_comment"], FILTER_SANITIZE_STRING);
```
Comment (sc979) [inline www/include/configuration/configObject/contactgroup/DB-Func.php:258]: same comment

File: www/include/configuration/configObject/contactgroup/DB-Func.php (hunk 1 above)
Comment (sc979) [reply www/include/configuration/configObject/contactgroup/DB-Func.php:209]: IMHO : you'll need to check that each of the sanitized variable are different from false

**Review by sc979 — CHANGES_REQUESTED** [review]

File: www/include/configuration/configObject/contactgroup/DB-Func.php [hunk 3]
```diff
@@ -236,19 +253,25 @@ function updateContactGroup($cg_id = null, $params = array())
         $ret = $form->getSubmitValues();
     }
 
-    $ret["cg_name"] = $centreon->checkIllegalChar($ret["cg_name"]);
+    $cgName = $centreon->checkIllegalChar(filter_var($ret["cg_name"], FILTER_SANITIZE_STRING));
+    $cgAlias = filter_var($ret["cg_alias"], FILTER_SANITIZE_STRING);
+    $cgComment = filter_var($ret["cg_comment"], FILTER_SANITIZE_STRING) ?: '';
```
Comment (loiclau) [inline www/include/configuration/configObject/contactgroup/DB-Func.php:258]:
```suggestion
    $cgComment = filter_var($ret["cg_comment"], FILTER_SANITIZE_STRING);
```

File: www/include/configuration/configObject/contactgroup/DB-Func.php [hunk 4]
```diff
@@ -190,26 +194,35 @@ function insertContactGroup($ret)
         $ret = $form->getSubmitValues();
     }
 
-    $ret["cg_name"] = $centreon->checkIllegalChar($ret["cg_name"]);
+    $cgName = $centreon->checkIllegalChar(filter_var($ret["cg_name"], FILTER_SANITIZE_STRING));
+    $cgAlias = filter_var($ret["cg_alias"], FILTER_SANITIZE_STRING);
+    $cgComment = filter_var($ret["cg_comment"], FILTER_SANITIZE_STRING);
+    $cgActivate = filter_var($ret["cg_activate"]["cg_activate"], FILTER_SANITIZE_STRING) ?: '1'; //enum
```
Comment (loiclau) [inline www/include/configuration/configObject/contactgroup/DB-Func.php:200]:
```suggestion
    $cgActivate = filter_var($ret["cg_activate"]["cg_activate"], FILTER_SANITIZE_STRING); //enum
    $cgActivate = ($cgActivate === '0' ? '0' : '1');
```

File: www/include/configuration/configObject/contactgroup/DB-Func.php [hunk 5]
```diff
@@ -236,19 +254,25 @@ function updateContactGroup($cg_id = null, $params = array())
...
```
Comment (loiclau) [inline www/include/configuration/configObject/contactgroup/DB-Func.php:260]:
```suggestion
    $cgActivate = filter_var($ret["cg_activate"]["cg_activate"], FILTER_SANITIZE_STRING); //enum
    $cgActivate = ($cgActivate === '0' ? '0' : '1');
```

File: www/include/configuration/configObject/contactgroup/DB-Func.php [hunk 6]
```diff
@@ -190,26 +194,36 @@ function insertContactGroup($ret)
         $ret = $form->getSubmitValues();
     }
 
-    $ret["cg_name"] = $centreon->checkIllegalChar($ret["cg_name"]);
+    $cgName = $centreon->checkIllegalChar(filter_var($ret["cg_name"], FILTER_SANITIZE_STRING));
+    $cgAlias = filter_var($ret["cg_alias"], FILTER_SANITIZE_STRING);
+    $cgComment = filter_var($ret["cg_comment"], FILTER_SANITIZE_STRING);
+    $cgActivate = filter_var($ret["cg_activate"]["cg_activate"], FILTER_SANITIZE_STRING); //enum
```
Comment (sc979) [inline www/include/configuration/configObject/contactgroup/DB-Func.php:200]: this line seems obsolete, as the enum will returns the state of the checkbox, which is checked on next line

File: www/include/configuration/configObject/contactgroup/DB-Func.php [hunk 7]
```diff
@@ -236,19 +254,26 @@ function updateContactGroup($cg_id = null, $params = array())
...
```
Comment (jeremyjaouen) [inline www/include/configuration/configObject/contactgroup/DB-Func.php:260]: https://github.com/centreon/centreon/pull/9220/files#r527052510

**Review by jeremyjaouen — DISMISSED** [review]

**Review by sc979 — DISMISSED** [review]

File: www/include/configuration/configObject/contactgroup/DB-Func.php [hunk 8]
```diff
@@ -236,19 +253,25 @@ function updateContactGroup($cg_id = null, $params = array())
         $ret = $form->getSubmitValues();
     }
 
-    $ret["cg_name"] = $centreon->checkIllegalChar($ret["cg_name"]);
+    $cgName = $centreon->checkIllegalChar(filter_var($ret["cg_name"], FILTER_SANITIZE_STRING));
+    $cgAlias = filter_var($ret["cg_alias"], FILTER_SANITIZE_STRING);
+    $cgComment = filter_var($ret["cg_comment"], FILTER_SANITIZE_STRING);
+    $cgActivate = ($ret["cg_activate"]["cg_activate"] === '0' ? '0' : '1');//enum
```
Comment (loiclau) [inline www/include/configuration/configObject/contactgroup/DB-Func.php:259]:
```suggestion
    $cgActivate = $ret["cg_activate"]["cg_activate"] === '0' ? '0' : '1';//enum
```

File: www/include/configuration/configObject/contactgroup/DB-Func.php [hunk 9]
```diff
@@ -190,26 +194,35 @@ function insertContactGroup($ret)
...
```
Comment (loiclau) [inline www/include/configuration/configObject/contactgroup/DB-Func.php:200]:
```suggestion
    $cgActivate = $ret["cg_activate"]["cg_activate"] === '0' ? '0' : '1';//enum
```

**Review by jeremyjaouen — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**